src/QMCHamiltonians/OperatorBase.h

```diff
@@ -62,30 +63,31 @@ struct NonLocalData : public QMCTraits
  * Return_t is defined as RealTye.
  * The types should be checked when using complex wave functions.
  */
-struct OperatorBase : public QMCTraits
+class OperatorBase : public QMCTraits
 {
-  /** type of return value of evaluate
-   */
-  typedef FullPrecRealType Return_t;
-  /** typedef for the serialized buffer
-   *
+public:
+  /// type of return value of evaluate
+  using Return_t = FullPrecRealType;
+  /**
+   * alias for the serialized buffer
    * PooledData<RealType> is used to serialized an anonymous buffer
    */
-  typedef ParticleSet::Buffer_t BufferType;
-  ///typedef for the walker
-  typedef ParticleSet::Walker_t Walker_t;
-  ///typedef for the ParticleScalar
-  typedef ParticleSet::Scalar_t ParticleScalar_t;
-
-  ///enum to denote energy domain of operators
-  enum energy_domains
+  using BufferType = ParticleSet::Buffer_t;
+  /// alias for the walker
+  using Walker_t = ParticleSet::Walker_t;
+  /// alias for the ParticleScalar
+  using ParticleScalar_t = ParticleSet::Scalar_t;
+
+  ///enum class to denote energy domain of operators
+  enum class energy_domains
```

Comment:
We have been using the convention for enum classes of,
```c++
enum class EnergyDomains 
{
KINETIC = 0,
POTENTIAL,
NO_ENERGY_DOMAIN
};
```
I thought this was in the developer section of the manual but it seems missing. Logic here is that an enum class is a type.  the All caps says this is an enum.